src/Engine/Combat/CombatContext.java

```diff
@@ -1,50 +1,140 @@
 package Engine.Combat;
 
+import java.util.ArrayList;
+import java.util.List;
+
+import Engine.Combat.StrikeParams.BattleParams;
+import Engine.UnitMods.UnitModifier;
 import Terrain.GameMap;
 import Units.Unit;
+import Units.UnitContext;
 import Units.WeaponModel;
 
 /**
  * CombatContext exists to allow COs to modify the fundamental parameters of an instance of combat.
  */
 public class CombatContext
 {
-  public Unit attacker, defender;
-  public WeaponModel attackerWeapon = null, defenderWeapon = null;
-  public int attackerX, attackerY, defenderX, defenderY;
-  public double attackerTerrainStars, defenderTerrainStars;
+  public UnitContext attacker, defender;
   public final GameMap gameMap; // for reference, not weirdness
   public boolean canCounter = false;
-  public boolean attackerMoved;
   public int battleRange;
-  
-  public CombatContext(GameMap map, Unit pAttacker, WeaponModel attackerWep, Unit pDefender, WeaponModel defenderWep, int pBattleRange, int attackerX, int attackerY)
+
+  public static CombatContext build(GameMap map,
+      Unit pAttacker, WeaponModel attackerWep, List<UnitModifier> attackerMods,
+      Unit pDefender, WeaponModel defenderWep, List<UnitModifier> defenderMods,
+      int pBattleRange, int attackerX, int attackerY)
+  {
+    CombatContext c = new CombatContext(map,
+        new UnitContext(map, pAttacker, attackerWep, attackerX, attackerY),
+        new UnitContext(map, pDefender, defenderWep, pDefender.x, pDefender.y),
+        pBattleRange);
+    c.attacker.mods.clear();
+    c.attacker.mods.addAll(attackerMods);
+    c.defender.mods.clear();
```

Comment:
Why is it necessary to clear the mods immediately after creating the objects and before adding the mods?

Reply:
I had it in my head that I was gonna support passing in a `UnitContext` straight from the top (thus pushing the `UnitContext` creation itself into an overload), and that it shouldn't assume the current mods were the right mods for combat consideration.
I don't think I planned out the API design well enough to know what assumptions I should be making.

Reply:
So then; do we need this, or can we remove the superfluous `clear`s?

Reply:
We don't. I deleted it entirely here: https://github.com/ThislsAUsername/DefendPeace/pull/346/commits/4c5377a89365d749c86db252c4487e5d259ecab6